libafl/src/inputs/wrapping.rs

```diff
@@ -0,0 +1,379 @@
+                    format!(
+                        "{:016x}",
+                        RandomState::with_seeds(0, 0, 0, 0).hash_one(self.as_ref())
+                    )
+                }
+            }
+
+            /// Input wrapping a <$t>
+            pub type $name = WrappingInput<$t>;
+        )*
+    };
+}
+
+// Invoke the macro with type-name pairs
+impl_input_for_wrapping_input!(
+    u8 => U8Input,
+    u16 => U16Input,
+    u32 => U32Input,
+    u64 => U64Input,
+    u128 => U128Input,
+    usize => UsizeInput,
+    i8 => I8Input,
+    i16 => I16Input,
+    i32 => I32Input,
+    i64 => I64Input,
+    i128 => I128Input,
+    isize => IsizeInput,
+);
+
+/// manually implemented because files can be written more efficiently
+impl Input for WrappingInput<Vec<u8>> {
+    fn generate_name(&self, _id: Option<CorpusId>) -> String {
+        format!(
+            "{:016x}",
+            RandomState::with_seeds(0, 0, 0, 0).hash_one(self.as_ref())
+        )
+    }
+
+    /// Write this input to the file
+    #[cfg(feature = "std")]
+    fn to_file<P>(&self, path: P) -> Result<(), Error>
+    where
+        P: AsRef<Path>,
+    {
+        write_file_atomic(path, self.as_ref())?;
+        Ok(())
+    }
+
+    /// Load the content of this input from a file
+    #[cfg(feature = "std")]
+    fn from_file<P>(path: P) -> Result<Self, Error>
+    where
+        P: AsRef<Path>,
+    {
+        let mut file = File::open(path)?;
+        let mut data = vec![];
+        file.read_to_end(&mut data)?;
+        Ok(data.into())
+    }
+}
+
+/// Constants for numeric types
+pub trait NumericConsts {
+    /// The number of bits, used for limiting shift operations
+    const BITS: u32;
+    /// The min value
+    const MIN: Self;
+    /// The max value
+    const MAX: Self;
+}
+
+macro_rules! impl_numeric_consts {
+    ( $( $t:ty ),* $(,)? ) => {
+        $(
+            impl NumericConsts for $t {
+                const BITS: u32 = <$t>::BITS;
+                const MIN: Self = <$t>::MIN;
+                const MAX: Self = <$t>::MAX;
+            }
+        )*
+    };
+}
+
+impl_numeric_consts!(u8, u16, u32, u64, u128, usize, i8, i16, i32, i64, i128, isize);
+
+impl<I, R> Shl<R> for WrappingInput<I>
+where
+    I: Shl<R>,
+    I::Output: Into<Self>,
+{
+    type Output = Self;
+
+    fn shl(self, rhs: R) -> Self::Output {
+        self.inner().shl(rhs).into()
+    }
+}
+
+impl<I> BitXorAssign for WrappingInput<I>
+where
+    I: BitXorAssign,
+{
+    fn bitxor_assign(&mut self, rhs: Self) {
+        self.as_mut().bitxor_assign(rhs.inner());
+    }
+}
+impl<I> BitOrAssign for WrappingInput<I>
+where
+    I: BitOrAssign,
+{
+    fn bitor_assign(&mut self, rhs: Self) {
+        self.as_mut().bitor_assign(rhs.inner());
+    }
+}
+
+impl<I> NumericConsts for WrappingInput<I>
+where
+    I: NumericConsts,
+{
+    const BITS: u32 = I::BITS;
+    const MIN: Self = WrappingInput::new(I::MIN);
+    const MAX: Self = WrappingInput::new(I::MAX);
```

Comment:
We can already use the `num_taits` crate for this I think, see:
https://docs.rs/num-traits/latest/num_traits/bounds/trait.Bounded.html#tymethod.max_value